addons/web/static/src/js/views/form_widgets.js

```diff
@@ -1137,17 +1137,22 @@ var FieldBinary = common.AbstractField.extend(common.ReinitializeFieldMixin, {
             var c = crash_manager;
             var filename_fieldname = this.node.attrs.filename;
             var filename_field = this.view.fields && this.view.fields[filename_fieldname];
+            var data = {
+                'model': this.view.dataset.model,
+                'id': this.view.datarecord.id,
+                'field': this.name,
+                'filename_field': filename_fieldname,
+                'download': true,
+            }
+            if (filename_field) {
+                data['filename'] = filename_field.get('value');
+            }
```

Comment:
What problem does this solve? `binary_content` seems to treat an empty filename or a filename of `None` the same way. Same question with the `data` key below.

Reply:
@xmo-odoo ```get_file``` uses FormData API which converts values to string so ```null``` was converted to string, so this will not pass ```filename``` and ```data``` key at all if it's false

Reply:
@msh-odoo 

> `null` was converted to `string`

To what string? An empty string or the non-empty string `"null"`?

Reply:
Hello @xmo-odoo 
In master: get_file uses Formdata API so, it converts the null value into "null" (string). so this will not pass filename and data key at all if it's false.

In 10.0: get_file does not use Formdata API but it makes jquery form. so, when there is no value then it leaves the value blank. so, null value converts into '' " (blank string).

Reply:
I see. Wouldn't it work just as well if we just replaced `: null` by `: ""` in the original code, so less code gets changed? The server seems to treat empty and absent filenames the same way, so far as I can tell.

Also still not understanding why the `data` key has been changed. I'm not sure it's even used at all anymore.

Reply:
@xmo-odoo Can you please have a look now? I have updated it as per your suggestion.